lib/vizkit/tree_view.rb

```diff
@@ -41,20 +55,30 @@ def tree_view.disconnected_items
         # warning: if the tree view is still visible it will reconnect
         # if a item gets expanded
         def tree_view.disconnect
-            0.upto(model.rowCount-1) do |i|
-                index = model.index(i,0)
-                next unless isExpanded(index)
-                model.item(i,0).collapse
-                model.item(i,1).collapse
+            if model.is_a?(Qt::AbstractProxyModel)
+                raw_model = model.sourceModel
+            else
+                raw_model = model
+            end
+            0.upto(raw_model.rowCount-1) do |i|
+                index = raw_model.index(i,0)
+                next unless isExpanded(model.is_a?(Qt::AbstractProxyModel) ? model.mapFromSource(index) : index)
+                raw_model.item(i,0).collapse
+                raw_model.item(i,1).collapse
                 disconnected_items << i
             end
         end
 
         # restores the state before disconnect was called
         def tree_view.reconnect
+            if model.is_a?(Qt::AbstractProxyModel)
+                raw_model = model.sourceModel
```

Comment:
Same. Make a `raw_model` method on `tree_view` that factors out this piece of code. (I would prefer `root_model` or `real_model`, but as long as you document the new method I don't mind `raw_model` at all)